code/frontend/src/components/OnboardingPage.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import './onboarding.css'; // Make sure to create this CSS file for styling
+
+const OnboardingPage = () => {
+  const [hobbies, setHobbies] = useState([]);
+  const [hobbyInput, setHobbyInput] = useState("");
+
+  // Function to handle the input change
+  const handleInputChange = (event) => {
+    setHobbyInput(event.target.value);
+  };
+
+  // Function to add hobby from input box
+  const addHobby = () => {
+    if (hobbyInput && !hobbies.includes(hobbyInput)) {
+      setHobbies([...hobbies, hobbyInput]);
+      setHobbyInput('');
+    }
+  };
+
+  // Function to add hobby from suggestions
+  const addSuggestedHobby = (suggestedHobby) => {
+    if (!hobbies.includes(suggestedHobby)) {
```

Comment:
convert the suggested hobby to lower case using `toLowerCase()` to avoid duplicates